go/inst/instance_dao.go

```diff
@@ -301,6 +301,8 @@ func ReadTopologyInstanceBufferable(instanceKey *InstanceKey, bufferWrites bool,
 	errorChan := make(chan error, 32)
 	var resolveErr error
 
+	instance.Key = *instanceKey
+
```

Comment:
Can you please clarify why you moved this line to this location? `if !instanceKey.IsValid()` is important context. If the key is invalid, we do not assign `instance.Key`.